python/lsst/analysis/ap/apdb.py

```diff
@@ -66,13 +78,65 @@ def connection(self):
         """
         pass
 
-    def load_sources_for_object(self, dia_object_id, limit=100000):
+    def set_bad_DiaSource_flags(self, flag_list):
```

Comment:
Maybe `set_source_exclude_flags`? "bad" has a variety of possible meanings, and these aren't "bad" per se, just "not included when you get data".